micro-rdk/src/esp32/conn/network.rs

```diff
@@ -299,6 +297,36 @@ impl Esp32WifiNetwork {
         log::info!("connection successful");
         Ok(())
     }
+    async fn try_connect_with_priority(
+        &self,
+        mut networks: Vec<NetworkSetting>,
+    ) -> Result<(), WifiManagerError> {
+        let access_points: Vec<String> = self
+            .scan_networks_inner()
+            .await
+            .inspect_err(|e| log::error!("failed to scan available networks: {}", e))?
+            .iter()
+            .map(|ap| ap.ssid.to_string())
+            .collect();
+        let access_points: HashSet<String> = HashSet::from_iter(access_points);
+        networks.sort_by(|a, b| b.priority.cmp(&a.priority));
```

Comment:
Are there allowed to be multiple networks with the same priority? If so, how should they sort? Should there be a secondary sort on the network name?

Reply:
added `impl Ord for NetworkSetting` that sorts by priority first, then name